### PR TITLE
fix(service): check if billing pricing value exists

### PR DESCRIPTION
**Pull request by antleblanc**
| Question         | Answer
| ---------------- | ---
| Branch?          | `master`
| Bug fix?         | yes
| New feature?     | no
| Breaking change? | no
| Tickets          | DTRSD-38569, DTRSD-39838
| License          | BSD 3-Clause

### :bug: Bug Fixes

161d60a - fix(service): check if billing pricing value exists

Signed-off-by: Antoine Leblanc <antoine.leblanc@ovhcloud.com>

### CODE REVIEW

**Review by JeremyDec — APPROVED** [review]

File: packages/manager/modules/models/service/Service.class.js [hunk 1]
```diff
@@ -103,7 +103,7 @@ export default class Service {
   isEngaged() {
     return (
       !!this.billing.engagement ||
-      !!this.billing.pricing.engagementConfiguration
+      !!(this.billing.pricing && this.billing.pricing.engagementConfiguration)
```
Comment (frenautvh) [inline packages/manager/modules/models/service/Service.class.js:106]:
```suggestion
      !!(this.billing.pricing?.engagementConfiguration)
```

Comment (antleblanc) [reply packages/manager/modules/models/service/Service.class.js:106]: You're right! But currently we are facing an issue with Rollup config that's why, as an Hotfix, I've picked this solution.

**Review by marie-j — APPROVED** [review]

File: packages/manager/modules/models/service/Service.class.js (hunk 1 above)
Comment (marie-j) [reply packages/manager/modules/models/service/Service.class.js:106]: Related to https://github.com/ovh/manager/issues/4099

Comment (antleblanc) [reply packages/manager/modules/models/service/Service.class.js:106]: #5064 solves this issue.
Still need some more investigation. We kill keep you posted.
Thank you!

**Pull request merged**